contracts/sidechain/AuraBalProxyOFT.sol

```diff
@@ -0,0 +1,326 @@
+    /// @param _authorized Whether to add or remove harvester
+    function updateAuthorizedHarvesters(address _harvester, bool _authorized) external onlyOwner {
+        authorizedHarvesters[_harvester] = _authorized;
+    }
+
+    /* -------------------------------------------------------------------
+       View functions 
+    ------------------------------------------------------------------- */
+
+    function circulatingSupply() public view override returns (uint256) {
+        return innerToken.totalSupply() - internalTotalSupply;
+    }
+
+    /* -------------------------------------------------------------------
+       Credit/Debit 
+    ------------------------------------------------------------------- */
+
+    /**
+     * @dev Override debtFrom to include a stake call after
+     * @param _from From address
+     * @param _srcChainId Source chain ID
+     * @param _toAddress Address to send to
+     * @param _amount Amount to send
+     */
+    function _debitFrom(
+        address _from,
+        uint16 _srcChainId,
+        bytes memory _toAddress,
+        uint256 _amount
+    ) internal override returns (uint256) {
+        uint256 amount = super._debitFrom(_from, _srcChainId, _toAddress, _amount);
+        _stakeAll();
+        internalTotalSupply += amount;
+        return amount;
+    }
+
+    /**
+     * @dev Override _creditTo to include a withdraw from vault before
+     * @param _srcChainId Source chain ID
+     * @param _toAddress Address to credit to
+     * @param _amount Amount to credit
+     */
+    function _creditTo(
+        uint16 _srcChainId,
+        address _toAddress,
+        uint256 _amount
+    ) internal override returns (uint256) {
+        _withdraw(_amount);
+        uint256 amount = super._creditTo(_srcChainId, _toAddress, _amount);
+        internalTotalSupply -= amount;
+        return amount;
+    }
+
+    /* -------------------------------------------------------------------
+       Harvest Rewards 
+    ------------------------------------------------------------------- */
+
+    /**
+     * @dev Harvest rewards from the compounder and distribute them to the source chains
+     *
+     *      Collect the amount of auraBAL that is staked on each source chain (L2). Then
+     *      trigger a harvest on the vault which calculates the amount of auraBAL that has
+     *      been earned since the last claim and the amount of extra rewards. These rewards
+     *      are then lazily distributed to the src chains proportionally.
+     *
+     *      Lazily meaning the claimable values are just added to a claimable mapping for
+     *      processing latest via processClaimable
+     *
+     * @param _srcChainIds Array of source chain layer zero IDs
+     * @param _totalUnderlying Array of totalUnderlying auraBAL staked on the source chain
+     * @param _totalUnderlyingSum Sum of values in _totalUnderlying array
+     */
+    function harvest(
```

Comment:
Calling more than once harvest without calling `processClaimable` can mess up the accountability of the claimable